sopel/bot.py

```diff
@@ -648,3 +646,71 @@ def cap_req(self, module_name, capability, arg=None, failure_callback=None,
             entry.append(_CapReq(prefix, module_name, failure_callback, arg,
                                  success_callback))
             self._cap_reqs[cap] = entry
+
+    def register_url_callback(self, pattern, callback):
+        """Register a ``callback`` for URLs matching the regex ``pattern``
+
+        :param pattern: compiled regex pattern to register
+        :param callback: callable object to handle matching URLs
+
+        .. versionadded: 7.0.0
+
+            This method has been added in Sopel 7.0.0 in order to offer
+            a unified interface for URL callbacks.
+
+        """
+        if not self.memory.contains('url_callbacks'):
+            self.memory['url_callbacks'] = tools.SopelMemory()
+        self.memory['url_callbacks'][pattern] = callback
+
+    def unregister_url_callback(self, pattern):
+        """Unregister the callback for URLs matching the regex ``pattern``
+
+        :param pattern: compiled regex pattern to unregister callback
+
+        .. versionadded: 7.0.0
+
+            This method has been added in Sopel 7.0.0 in order to offer
+            a unified interface to URL callbacks.
+
+        """
+        if not self.memory.contains('url_callbacks'):
+            # nothing to unregister
+            return
+
+        try:
+            del self.memory['url_callbacks'][pattern]
+        except KeyError:
+            pass
+
+    def search_url_callbacks(self, url):
+        """Yield callbacks found for ``url`` matching their regex pattern
+
+        :param str url: URL found in a trigger
+        :return: yield 2-value tuples of ``(callback, match)``
+
+        For each pattern that matches the ``url`` parameter, it yields a
+        2-value tuple of ``(callable, match)`` for that pattern.
+
+        The ``callable`` is the one registered with
+        :meth:`register_url_callback`, and the ``match`` is the result of
+        the regex pattern's ``search`` method.
+
+        .. versionadded: 7.0.0
+
+            This method has been added in Sopel 7.0.0 in order to offer
+            a unified interface for URL callbacks.
```

Comment:
Not sure what would be more useful to put here, but the comment I made about redundancy on `register_` above applies here. Maybe this one doesn't need an explanation for the `versionadded` directive at all?

Reply:
Removed explanation - done.